lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -19,7 +19,8 @@ sub _init {
     my ($self) = @_;
 
     return 0 unless my $job = $self->app->schema->resultset('Jobs')->find($self->param('testid'));
-    my $module = $job->modules->search({name => $self->param('moduleid')})->first;
+    my %attrs = (rows => 1, order_by => {-desc => 'id'});
+    my $module = $job->modules->search({name => $self->param('moduleid')}, \%attrs)->first;
```

Comment:
Since this is limited to one row I suppose one could also use `find` here.

Reply:
I looked at this in my PR, too, but the `name` is not a primary or unique key, so theoretically this could return more than one row and issue a warning.

Reply:
But it specifies `rows => 1` as attributes.